### PR TITLE
Changed bare excepts in ode.py

### LINKED ISSUE #11668: Get rid of bare "except"s
It looks like quite a few bare `except:` and `except Exception:` lines have crept up throughout the codebase. A few of these are OK (for instance, the ones in `runtests.py` are there to catch and report exceptions in the tests), but most should be replaced with an explicit exception, or rearranged so that the logic does not require checking for an exception. Even places where you really do want to catch all exceptions, `except Exception:` is usually better than `except:`, since the latter also catches things like `KeyboardInterrupt` and `MemoryError`, which you generally really do want to raise to the user. I didn't check all the instances in the code, but I can't think of an example where bare `except:` would be acceptable, excepting the case where there is a `raise` line in the except block. 

I've marked this as easy to fix. It does require a bit of work, however, as you'll need to understand what the code is doing in each instance to know how to fix it (or if it shouldn't be fixed).

### PR DESCRIPTION
**Pull request by ghost**
<!-- If this pull request fixes an issue please indicate which issue by typing "Fixes #NNNN" below.>

"Helps fix #11668"

### CODE REVIEW

File: sympy/solvers/ode.py [hunk 1]
```diff
@@ -8122,15 +8122,15 @@ def _nonlinear_3eq_order1_type2(x, y, z, t, eq):
     y_z = sqrt(((a*C1-C2) - c*(a-c)*z(t)**2)/(b*(a-b)))
     try:
         sol1 = dsolve(a*diff(x(t),t) - (b-c)*y_x*z_x*r[f]).rhs
-    except:
+    except Exception:
```
Comment (gxyd) [inline sympy/solvers/ode.py]: You should specific in the type of `Exception` you want to raise. So is the case below.

**Comment by gxyd** [comment]
@OppenheimerAndTheMartians can you please address the requested changes?

**Comment by OppenheimerAndTheMartians** [comment]
They look fine. I'm sorry I haven't been able to contribute to the project as I thought I'd be able to 6 months ago.
 

Sent: Friday, October 06, 2017 at 1:00 PM

**Comment by gxyd** [comment]
I am closing this pull request for now, since this PR has been inactive. Please feel free to push code addressing the requested changes, and then re-open this PR or open a new PR (in case GitHub doesn't allow to reopen this PR).

Also none of the changes here right now is correct.

**Pull request closed without merging**